### PR TITLE
Add new Roborock integration documentation

### PR DESCRIPTION
**Pull request by Lash-L**
## Proposed change

This is a conversion of the popular hacs integration: https://github.com/humbertogontijo/homeassistant-roborock. It aims to allow users to be able to use the Roborock app and the home assistant integration, rather than having to use the Xiaomi app.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
  - [ ] I've opened up a PR to add logo's and icons in [Brands repository](https://github.com/home-assistant/brands).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase:  https://github.com/home-assistant/core/pull/89456
- Link to parent pull request in the Brands repository: 
- This PR fixes or closes issue: fixes #

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards].

[standards]: https://developers.home-assistant.io/docs/documenting/standards

### CODE REVIEW

File: source/_integrations/roborock.markdown
Comment (frenck) [inline source/_integrations/roborock.markdown]:
```suggestion
The Roborock integration allows you to control your Roborock](https://us.roborock.com/pages/robot-vacuum-cleaner) vacuum.
```

**Review by frenck — CHANGES_REQUESTED** [review]
Since a Roborock can also be controlled fully locally using the Xiaomi  integration, it might be nice to add some more information in the introduction text.

**Comment by Lash-L** [comment]
> Since a Roborock can also be controlled fully locally using the Xiaomi integration, it might be nice to add some more information in the introduction text.

Agreed. I will try to go through and add some more info later tonight. Thank you! I plan to try to figure out local control down the line, but it is not in the hacs integration, so it will be a bit before I get to it as I figured it made more sense to add all of those features first.

**Comment by Lash-L** [comment]
> Since a Roborock can also be controlled fully locally using the Xiaomi integration, it might be nice to add some more information in the introduction text.

Added a bit more information. More than happy to add more if needed

File: source/_integrations/roborock.markdown
Comment (rytilahti) [inline source/_integrations/roborock.markdown]:
```suggestion
description: Instructions on how to integrate Roborock vacuums into Home Assistant
```
I'm not sure where/if this string is used, but this should likely be in plural form?

File: source/_integrations/roborock.markdown
Comment (rytilahti) [inline source/_integrations/roborock.markdown]:
```suggestion
The Roborock integration allows you to control your Roborock](https://us.roborock.com/pages/robot-vacuum-cleaner) vacuum while keeping using the Roborock app. In contrast to [Xiaomi Miio](https://www.home-assistant.io/integrations/xiaomi_miio/) integration, this integration provides more complete support but requires cloud access and device configuration using the Roborock app.
```
Just a suggestion to reword this a bit to clearly point out the strengths & weaknesses compared to the existing integration.

I have learned that it is also a good idea not to add todo-like (word "currently") information to the documentation :-)

**Review by rytilahti — COMMENTED** [review]
I think it's great to get more complete support for roborock's into homeassistant, thanks @Lash-L! Added a couple of suggestions on the wording (with my roborock-using xiaomi_miio maintainer hat on) not having tested the PR itself.

**Comment by Lash-L** [comment]
> I think it's great to get more complete support for roborock's into homeassistant, thanks @Lash-L! Added a couple of suggestions on the wording (with my roborock-using xiaomi_miio maintainer hat on) not having tested the PR itself.

Thank you! I agreed with your suggestions and I went ahead and made them.

Thank you for your feedback and thank you for all of your work on the Xiaomi miio integration!

(And if your roborock-using xiaomii_miio maintainer self ever want to contribute anything on this project, you are more than welcome!)

File: source/_integrations/roborock.markdown
Comment (frenck) [inline source/_integrations/roborock.markdown]:
```suggestion
The Roborock integration allows you to control your [Roborock](https://us.roborock.com/pages/robot-vacuum-cleaner) vacuum while continuing to use the Roborock app.

In contrast to [Xiaomi Miio](/integrations/xiaomi_miio/) integration, this integration provides more complete support but requires cloud access and device configuration using the Roborock app.
```

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.
⚠️ The `ha_release` might need an update before merging this one in.

**Comment by Lash-L** [comment]
Appreciate you @frenck! Thanks

File: source/_integrations/roborock.markdown
Comment (frenck) [inline source/_integrations/roborock.markdown]:
```suggestion
In contrast to [Xiaomi Miio](/integrations/xiaomi_miio/) integration, this integration provides more complete support but requires cloud access to set up and device configuration using the Roborock app.

Once you log in with your Roborock account, the integration will automatically discover your Roborock devices and get the needed information to communicate locally with them. Please ensure your Home Assistant instance can communicate with the local IP of your device. We recommend setting a static IP for your Roborock Vacuum to help prevent future issues.
```

**Review by frenck — APPROVED** [review]
Thanks, @Lash-L 👍 

../Frenck

**Comment by Lash-L** [comment]
Thanks frenck!

**Pull request merged**
